sed/loader/flash/loader.py

```diff
@@ -0,0 +1,615 @@
+        [train_id, np_array] = self.create_numpy_array_per_channel(
+            h5_file,
+            channel,
+        )  # numpy Array created
+        channel_dict = self._config["dataframe"]["channels"][
+            channel
+        ]  # channel parameters
+
+        # If np_array is size zero, fill with NaNs
+        if np_array.size == 0:
+            np_array = np.full_like(train_id, np.nan, dtype=np.double)
+            data = Series(
+                (np_array[i] for i in train_id.index),
+                name=channel,
+                index=train_id,
+            )
+
+        # Electron resolved data is treated here
+        if channel_dict["format"] == "per_electron":
+            # Creates the index_per_electron if it does not exist for a given file
+            if self.index_per_electron is None:
+                self.create_multi_index_per_electron(h5_file)
+
+            data = self.create_dataframe_per_electron(
+                np_array,
+                train_id,
+                channel,
+            )
+
+        # Pulse resolved data is treated here
+        elif channel_dict["format"] == "per_pulse":
+            data = self.create_dataframe_per_pulse(
+                np_array,
+                train_id,
+                channel,
+                channel_dict,
+            )
+
+        # Train resolved data is treated here
+        elif channel_dict["format"] == "per_train":
+            data = self.create_dataframe_per_train(np_array, train_id, channel)
+
+        else:
+            raise ValueError(
+                channel
+                + "has an undefined format. Available formats are \
+                per_pulse, per_electron and per_train",
+            )
+
+        return data
+
+    def concatenate_channels(
+        self,
+        h5_file: h5py.File,
+    ) -> Union[Series, DataFrame]:
+        """Returns a concatenated pandas DataFrame for either all pulse,
+        train or electron resolved channels."""
+        all_keys = parse_h5_keys(h5_file)  # Parses all channels present
+
+        # Check for if the provided group_name actually exists in the file
+        for channel in self._config["dataframe"]["channels"]:
+            if channel == "timeStamp":
+                group_name = (
+                    self._config["dataframe"]["channels"][channel][
+                        "group_name"
+                    ]
+                    + "time"
+                )
+            else:
+                group_name = (
+                    self._config["dataframe"]["channels"][channel][
+                        "group_name"
+                    ]
+                    + "value"
+                )
+
+            if group_name not in all_keys:
+                raise ValueError(
+                    f"The group_name for channel {channel} does not exist.",
+                )
+
+        data_frames = (
+            self.create_dataframe_per_channel(h5_file, each)
+            for each in self.available_channels
```

Comment:
I suppose this mostly applies to the Aux channels. Maybe I misread the code. I was just puzzled, why, e.g. you output the DLDBinSize for every electron...

Reply:
yes, this comes from the data structure. We could hard code ignoring these an loading them differently, but as in comments above, this would be a possible performance improvement we can do if necessary.